tests/cases/conformance/jsdoc/checkJsdocTypeTag5.ts

```diff
@@ -16,3 +16,23 @@ function i(x) { return x }
 var j = x => x
 /** @type {{ (x: number): string }} */
 var k = function (x) { return x }
+
+
+/** @typedef {(x: 'hi' | 'bye') => 0 | 1 | 2} Argle */
+/** @type {Argle} */
+function blargle(s) {
+    return 0;
+}
+
+/** @type {0 | 1 | 2} - assignment should not error */
+var zeroonetwo = blargle('hi')
+
+/** @typedef {{(s: string): 0 | 1; (b: boolean): 2 | 3 }} Gioconda */
+
+/** @type {Gioconda} */
+function monaLisa(sb) {
+    return typeof sb === 'string' ? 1 : 2;
+}
+
+/** @type {2 | 3} - overloads are not supported, so there will be an error */
+var twothree = monaLisa(false);
```

Comment:
The error here doesn't say anything about overloads through. (`error TS2322: Type '1 | 2' is not assignable to type '2 | 3'.`) The inferred return type of the function `1 | 2` is used, but it really shouldn't if there was an explicit type annotation. If the function returned `any`, there would be silent failure here, even though it had an explicit type annotation.

Reply:
You're right, it's only an error because we go back to ignoring the type annotation. The correct thing is to treat a type annotation with multiple signatures as an overload set and use the normal typescript rules for checking that the implementation signature is assignable to the overload set. 

But the JS implementation isn't typed, so that check doesn't really make sense. Basically, I left it the way it was because I couldn't think of any useful checking.

Reply:
After discussing this in person, here's what we decided

1. `/** @type {number} */` should error. It doesn't today, but I think it's clearly wrong *enough* to deserve an error.
2. `/** @type {Gioconda} */` should give `monaLisa` an overload type, and the function parameters and return type should be checked according to the typescript overload rules. This will usually succeed since the parameters will be `any`, and the return type will therefore usually be `any`. But it will give the desired overload type to callers of the function, which is why somebody would write this type in the first place.
3. (2) is not just a simple fix so I will leave the current silent-fallback behaviour in this PR and open an issue to track it.

Reply:
Opened #25590 to track (2).